examples/showcase/eclipse_amount.py

```diff
@@ -0,0 +1,66 @@
+"""
+===================================
+Obtaining solar-eclipse information
+===================================
+
+How to obtain information about a solar eclipse
+
+Here we look into the total solar eclipse on 2024 April 8, which passes over
+Dallas, Texas.
+
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.dates import DateFormatter
+
+import astropy.units as u
+from astropy.coordinates import EarthLocation, solar_system_ephemeris
+from astropy.time import Time
+
+from sunpy.coordinates import sun
+
+##############################################################################
+# Create an array of observation times centered around the maximum of total
+# solar eclipse at ~18:43 UTC.  We sample per second within 5 minutes of
+# maximum, and per minute further out.
+
+obstime = (Time('2024-04-08 18:43')
+           + np.concatenate([np.arange(-100, -5) * u.min,
+                             np.arange(-300, 300) * u.s,
+                             np.arange(5, 100) * u.min]))
+
+##############################################################################
+# Create an observer coordinate for the longitude and latitude of Dallas for
+# the array of observation times.
+
+location = EarthLocation.from_geodetic(-96.808891*u.deg, 32.779167*u.deg)
+observer = location.get_itrs(obstime)
+
+##############################################################################
+# Calculate the eclipse amount using a JPL ephemeris.
+
+with solar_system_ephemeris.set('de432s'):
+    amount = sun.eclipse_amount(observer)
+    amount_minimum = sun.eclipse_amount(observer, minimum=True)
+
+##############################################################################
+# Calculate the start and end points of total solar eclipse.
+
+start_part, end_part = obstime[np.flatnonzero(amount > 0)[[0, -1]]]
+start_total, end_total = obstime[np.flatnonzero(amount_minimum == 1)[[0, -1]]]
+print(f"Partial solar eclipse starts at {start_part}\n"
+      f"Total solar eclipse starts at {start_total}\n"
+      f"Total solar eclipse ends at {end_total}\n"
+      f"Partial solar eclipse ends at {end_part}")
+
+##############################################################################
+# Plot the eclipse amount.
+
+plt.figure()
+plt.plot(obstime.datetime64, amount)
+plt.gca().xaxis.set_major_formatter(DateFormatter('%H:%M'))
```

Comment:
Is there a reason to not use `astropy.visualization.time_support` and just plot `obstime`?

Reply:
I was probably doing something dumb, but I could not set the formatting the way I wanted when using `time_support()`.  Feel free to "fix" my code!

Reply:
After playing with this a lot more, I've come to the conclusion that `time_support()` is worthless when the desire is to plot in "normal" time (i.e., hours/minutes/seconds).  You're going to want to format it using `DateFormatter` anyway, and if you want to use a time zone, it's arguably negative value to use `time_support()` because it automatically tacks on an axis label that is not correct.